tests/catch/unit/stream/hipStreamValue.cc

```diff
@@ -0,0 +1,495 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+#include <hip_test_common.hh>
+
+// Random predefiend 32 and 64 bit values
+constexpr uint32_t value32 = 0x70F0F0FF;
+constexpr uint64_t value64 = 0x7FFF0000FFFF0000;
+constexpr unsigned int writeFlag = 0;
+
+constexpr uint32_t DATA_INIT = 0x1234;
+constexpr uint32_t DATA_UPDATE = 0X4321;
+
+template <typename intT> struct TEST_WAIT {
+  using uintT = typename std::make_unsigned<intT>::type;
+  int compareOp;
+  uintT mask;
+  uintT waitValue;
+  intT signalValueFail;
+  intT signalValuePass;
+
+  TEST_WAIT(int compareOp, uintT waitValue, intT signalValueFail, intT signalValuePass)
+      : compareOp{compareOp},
+        waitValue{waitValue},
+        signalValueFail{signalValueFail},
+        signalValuePass{signalValuePass} {
+    mask = static_cast<uintT>(0xFFFFFFFFFFFFFFFF);
+  }
+
+  TEST_WAIT(int compareOp, uintT mask, uintT waitValue, intT signalValueFail, intT signalValuePass)
+      : compareOp{compareOp},
+        waitValue{waitValue},
+        signalValueFail{signalValueFail},
+        signalValuePass{signalValuePass},
+        mask{mask} {}
+};
+typedef TEST_WAIT<int32_t> TEST_WAIT32;
+typedef TEST_WAIT<int64_t> TEST_WAIT64;
+
+bool streamWaitValueSupported() {
+  int device_num = 0;
+  HIP_CHECK(hipGetDeviceCount(&device_num));
+  int waitValueSupport;
+  for (int device_id = 0; device_id < device_num; ++device_id) {
+    HIP_CHECK(hipSetDevice(device_id));
+    waitValueSupport = 0;
+    HIP_CHECK(hipDeviceGetAttribute(&waitValueSupport, hipDeviceAttributeCanUseStreamWaitValue,
+                                    device_id));
+    if (waitValueSupport == 1) return true;
+  }
+  return false;
+}
+
+// hipStreamWriteValue Tests
+TEST_CASE("Unit_hipStreamValue_Write") {
+  int64_t* signalPtr;
+
+  hipStream_t stream{nullptr};
+  HIP_CHECK(hipStreamCreate(&stream));
+
+  // Allocate Host Memory
+  auto hostPtr64 = std::unique_ptr<uint64_t>(new uint64_t(1));
+  auto hostPtr32 = std::unique_ptr<uint32_t>(new uint32_t(1));
+
+  // Register Host Memory
+  HIP_CHECK(hipHostRegister(hostPtr64.get(), sizeof(int64_t), 0));
+  HIP_CHECK(hipHostRegister(hostPtr32.get(), sizeof(int32_t), 0));
+
+  // Register Signal Memory
+  HIP_CHECK(hipExtMallocWithFlags((void**)&signalPtr, 8, hipMallocSignalMemory));
+
+  // Initialise Data
+  *signalPtr = 0x0;
+  *hostPtr64 = 0x0;
+  *hostPtr32 = 0x0;
+
+  SECTION("Registered host memory hipStreamWriteValue32") {
+    INFO("Test writting to registered host pointer using hipStreamWriteValue32");
+    HIP_CHECK(hipStreamWriteValue32(stream, hostPtr32.get(), value32, writeFlag));
+    HIP_CHECK(hipStreamSynchronize(stream));
+    HIP_ASSERT(*hostPtr32 == value32);
+  }
+
+  SECTION("Registered host memory hipStreamWriteValue64") {
+    INFO("Test writting to registered host pointer using hipStreamWriteValue32");
+    HIP_CHECK(hipStreamWriteValue64(stream, hostPtr64.get(), value64, writeFlag));
```

Comment:
Request to please add test cases to verify hipStreamWriteValue64/hipStreamWriteValue32 for arrays. 
Need to test memory with offset

Reply:
Are you requesting for there to be additional tests for array inputs **and** offset inputs? Or offset array inputs?

Reply:
@agunashe Do you have an answer for Dylan?

Reply:
@dylan-angus-codeplay - need test cases for array contiguous memory as input both for hipStreamWriteValue64/hipStreamWriteValue32 and hipStreamWaitValue64/hipStreamWaitValue32

Reply:
@agunashe @chrispaquot I have created an internal ticket to write the tests requested. 
If it's acceptable to you, this PR could still be merged and we will create a new PR for the array tests.